### ISSUE TITLE
Ability to exclude diff signs from "changes" copied contents

### ISSUE THREAD

**Issue by teocns** [jesseduffield/lazygit#3859]
**Is your feature request related to a problem? Please describe.**

Copying (`<C-o>`) from the "changes" panel adds `+` and `-` symbols on the first column of each line as diff contents are copied.

**Describe the solution you'd like**

To optionally enable / disable including diff signs in the copy buffer, or the ability to bind such filtered copy mode to an other key-mapping different than `<C-o>`

**Comment by 7flash** [comment]
I need this because currently its a basis of my development workflow, specifically when looking at ai-generated commits in my codebase, I begin to review the diff from the end of the file, selecting relevant lines to clipboard (Ctrl+O) then switching back and forth between lazygit and helix editor (Cmd+`)  where I jump to the line mentioned in diff (g+number) and replace it (Shift+X d Cmd+v)

Then, to get rid of + symbol, I have to do additionally (Shift+X s Enter d), and I hope this enhancement can relieve me from this additional step.

**Comment by stefanhaller** [comment]
I've needed this too a few times, and found it annoying to have to manually remove the +/- characters.

However, I wonder if it isn't confusing when you select a hunk of both selected and deleted lines, and then you can no longer tell which was which after pasting. I wonder if we should drop the +/- characters only when all selected lines have the same one, but keep them otherwise. What do people think?

@teocns @7flash @Rpinski @Kirow

**Comment by Rpinski** [comment]
Good thought, @stefanhaller 👍

That would be a clever solution. How about the (maybe rare) case when you copy changed and unchanged lines together (like unchanged and added)? I would expect the "+"es to removed here, too, but maybe other people would expect them to stay?

**Comment by stefanhaller** [comment]
@Rpinski Yes, I had the same thought, I think it's reasonable to drop them in that case, too.

**Comment by stefanhaller** [comment]
Here's a PR, can you please test if this addresses our needs? #4519

**Comment by Rpinski** [comment]
It looks great to me so far 👍

**Comment by itdhsc** [comment]
Great work finding a solution. Looking forward for the new update 🚀
